File: app/src/main/java/com/example/rpicommunicator_v1/Plants/PlantAdapter.java
package com.example.rpicommunicator_v1.Plants;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.rpicommunicator_v1.Database.Plant;
import com.example.rpicommunicator_v1.R;
import com.example.rpicommunicator_v1.ViewAndModels.MainActivityViewModel;

import java.util.ArrayList;
import java.util.List;


public class PlantAdapter extends RecyclerView.Adapter<PlantAdapter.PlantHolder> {
    private List<Plant> plants = new ArrayList<>();
    private OnItemClickListener mListener;
    private OnItemLongClickListener mlongListener;
    private MainActivityViewModel mainActivityViewModel;


    public void setViewModel(MainActivityViewModel mainActivityViewModel) {
        this.mainActivityViewModel = mainActivityViewModel;
    }

    public interface OnItemClickListener {
        void onItemClicked(int position);
    }

    public interface OnItemLongClickListener {
        void onItemClicked(int position);
    }

    public void setOnItemClickListener(OnItemClickListener listener) {
        mListener = listener;
    }

    public void setOnItemLongClickListener(OnItemLongClickListener longListener) {
        this.mlongListener = longListener;
    }


    @Override
    public PlantHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.plant_item, parent, false);
        PlantHolder evh = new PlantHolder(v, mListener, mlongListener);
        return evh;
    }


    @Override
    public void onBindViewHolder(PlantHolder holder, int position) {
        Plant currentItem = plants.get(position);

        if (currentItem.getImageID() != -1) {
            holder.mImageView.setImageResource(currentItem.getIconID());
            float alpha = (float) 1;
            holder.mImageView.setAlpha(alpha);
        } else {
            //holder.blurHelper.setCardBackgroundColor(R.color.dark_yellow);//setBackgroundColor(R.color.dark_yellow);
            holder.mImageView.setImageResource(R.drawable.icon_pump);
            float alpha = (float) 0.1;
            holder.mImageView.setAlpha(alpha);
        }
        holder.mTextView1.setText(currentItem.getName());
        holder.mTextView2.setText(currentItem.getInfo());
        holder.datumView.setText(currentItem.getWatered());
        holder.button_delete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                v.setVisibility(View.GONE);
                mainActivityViewModel.remove(plants.get(position));

            }
        });
    }

    @Override
    public int getItemCount() {
        return plants.size();
    }

    public void setPlants(List<Plant> plants) {
        this.plants = plants;
        notifyDataSetChanged();
    }


    public static class PlantHolder extends RecyclerView.ViewHolder {
        public ImageView mImageView;
        public TextView mTextView1;
        public TextView mTextView2;
        public TextView datumView;
        public Button button_delete;

        public PlantHolder(View itemView, final OnItemClickListener listener, final OnItemLongClickListener longListener) {
            super(itemView);
            mImageView = itemView.findViewById(R.id.imageView);
            mTextView1 = itemView.findViewById(R.id.title);
            mTextView2 = itemView.findViewById(R.id.textView);
            datumView = itemView.findViewById(R.id.textView2);
            button_delete = itemView.findViewById(R.id.delete_button);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (listener != null) {
                        int position = getAdapterPosition();
                        if (position != RecyclerView.NO_POSITION) {
                            listener.onItemClicked(position);
                        }
                    }
                }
            });

            itemView.setOnLongClickListener(new View.OnLongClickListener() {
                @Override
                public boolean onLongClick(View v) {
                    if (listener != null) {
                        int position = getAdapterPosition();
                        if (position != RecyclerView.NO_POSITION) {

                            if (button_delete.getVisibility()==View.GONE)
                            {
                                button_delete.setVisibility(View.VISIBLE);
                            }else {
                                button_delete.setVisibility(View.GONE);
                            }
                            return true;
                        }
                    }
                    return false;
                }
            });


        }
    }

}

File: app/src/main/java/com/example/rpicommunicator_v1/Plants/PlantAdapter.kt
package com.example.rpicommunicator_v1.Plants

import androidx.recyclerview.widget.RecyclerView
import com.example.rpicommunicator_v1.Plants.PlantAdapter.PlantHolder
import com.example.rpicommunicator_v1.Database.Plant.Plant
import com.example.rpicommunicator_v1.ViewAndModels.MainActivityViewModel
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.example.rpicommunicator_v1.R
import android.widget.TextView
import android.view.View.OnLongClickListener
import android.widget.Button
import android.widget.ImageView
import java.util.ArrayList

class PlantAdapter : RecyclerView.Adapter<PlantHolder>() {

    private var plants: List<Plant> = ArrayList()
    private lateinit var mListener:  (View, Int, Int)-> Unit;
    private var mlongListener: OnItemLongClickListener? = null
    private var mainActivityViewModel: MainActivityViewModel? = null


    fun setViewModel(mainActivityViewModel: MainActivityViewModel?) {
        this.mainActivityViewModel = mainActivityViewModel
    }

    interface OnItemClickListener {
        fun onItemClicked(position: Int)
    }

    interface OnItemLongClickListener {
        fun onItemClicked(position: Int)
    }

    fun setOnItemClickListener(listener: (View, Int, Int) -> Unit) {
        mListener = listener
    }

    fun setOnItemLongClickListener(longListener: OnItemLongClickListener?) {
        mlongListener = longListener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PlantHolder {
        val v =
            LayoutInflater.from(parent.context).inflate(R.layout.plant_item, parent, false)
        return PlantHolder(v, mListener, mlongListener)
    }

    override fun onBindViewHolder(holder: PlantHolder, position: Int) {
        val currentItem = plants[position]
        if (currentItem.imageID != -1) {
            holder.mImageView.setImageResource(currentItem.iconID)
            val alpha = 1.toFloat()
            holder.mImageView.alpha = alpha
        } else {
            //holder.blurHelper.setCardBackgroundColor(R.color.dark_yellow);//setBackgroundColor(R.color.dark_yellow);
            holder.mImageView.setImageResource(R.drawable.icon_pump)
            val alpha = 0.1.toFloat()
            holder.mImageView.alpha = alpha
        }
        holder.mTextView1.text = currentItem.name
        holder.mTextView2.text = currentItem.info
        holder.datumView.text = currentItem.watered
        holder.button_delete.setOnClickListener { v ->
            v.visibility = View.GONE
            mainActivityViewModel!!.remove(plants[position])
        }
    }

    override fun getItemCount(): Int {
        return plants.size
    }

    fun setPlants(plants: List<Plant>) {
        this.plants = plants
        notifyDataSetChanged()
    }

    class PlantHolder(
        itemView: View,
        listener: (View, Int, Int) -> Unit,
        longListener: OnItemLongClickListener?
    ) : RecyclerView.ViewHolder(itemView) {
        var mImageView: ImageView
        var mTextView1: TextView
        var mTextView2: TextView
        var datumView: TextView
        var button_delete: Button

        init {
            mImageView = itemView.findViewById(R.id.imageView)
            mTextView1 = itemView.findViewById(R.id.title)
            mTextView2 = itemView.findViewById(R.id.textView)
            datumView = itemView.findViewById(R.id.textView2)
            button_delete = itemView.findViewById(R.id.delete_button)
            itemView.setOnClickListener {
                val position = bindingAdapterPosition
                if (position != RecyclerView.NO_POSITION) {
                    //listener.onItemClicked(position)
                    //listener.
                    listener

                            //idee: listener lambda ausführen
                }
            }
            itemView.setOnLongClickListener(OnLongClickListener {
                if (listener != null) {
                    val position = adapterPosition
                    if (position != RecyclerView.NO_POSITION) {
                        if (button_delete.visibility == View.GONE) {
                            button_delete.visibility = View.VISIBLE
                        } else {
                            button_delete.visibility = View.GONE
                        }
                        return@OnLongClickListener true
                    }
                }
                false
            })
        }
    }
}